storage/shard.go

```diff
@@ -219,6 +219,8 @@ type dbShardEntryWorkFn func(entry *dbShardEntry) bool
 
 type dbShardEntryBatchWorkFn func(entries []*dbShardEntry) bool
 
+type listElement *list.Element
```

Comment:
suggestion(take/leave): rename to `shardListElement` to be explicit about how the type is used.